### PR TITLE
Unit test

### PR DESCRIPTION
**Pull request by boniface193**
### we have done the following to reach this milestone:

- [ ] Added unit tests for all the methods in all the classes:
    - [ ] Book
    - [ ] Person
    - [ ] Student
    - [ ] Teacher
    - [ ] Classroom
    - [ ] Rental

### CODE REVIEW

File: spec/app_spec.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    end
+  end
+  # Add tests for the remaining methods in a similar manner
+  describe '#add_book' do
+    it 'should add a book' do
+      allow(app).to receive(:gets).and_return('Book 1', 'Author 1')
+      expect { app.add_book }.to output(/Book created successfully/).to_stdout
+    end
+  end
+end
```
Comment (Whoistolu) [inline spec/app_spec.rb]: - [ ] [REQUIRED] Hello Team, you have done a great job by adding unit tests for most of the methods in the class, however,  as stated by the project requirement we still need to add unit test for the following methods:
`add_rental`, `list_rentals_for_person` 

For example:
``` ruby
describe '#add_rental' do
  it 'should add a rental' do
    book = Book.new('Book 1', 'Author 1')
    student = Student.new(20, 'Student 1', parent_permission: true)
    app.books << book
    app.people << student

    allow(app).to receive(:gets).and_return('0', '0', '2022-09-15')

    expect { app.add_rental }.to output(/Rental created successfully/).to_stdout
    expect(app.rentals).to include(an_instance_of(Rental))
  end
end

describe '#list_rentals_for_person' do
  it 'should list rentals for a person' do
    book = Book.new('Book 1', 'Author 1')
    student = Student.new(20, 'Student 1', parent_permission: true)
    rental = Rental.new('2022-09-15', book, student)
    app.books << book
    app.people << student
    app.rentals << rental

    allow(app).to receive(:gets).and_return('1')

    expect { app.list_rentals_for_person }.to output(/Date: 2022-09-15, Book "Book 1" by Author 1/).to_stdout
  end
end
```

File: spec/book_spec.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    it 'should create a new book with the given title and author' do
+      expect(book.title).to eq('Test Book')
+      expect(book.author).to eq('Test Author')
+    end
+
+    it 'initialize an empty rental array' do
+      expect(book.rentals).to be_empty
+    end
+  end
+end
```
Comment (Whoistolu) [inline spec/book_spec.rb]: - [ ] [REQUIRED] The given tests focus on the initialization of the Book class and checking if the rentals array is empty after initialization. As recommended above, let's also add some unit tests for the `add_rental ` method of the `book` class For example :

``` ruby
describe '#add_rentals' do
  it 'should add a new rental to the rentals array' do
    person = instance_double('Person')
    date = '2022-09-15'

    expect { book.add_rentals(date, person) }.to change { book.rentals.size }.by(1)
    expect(book.rentals.last.date).to eq('2022-09-15')
    expect(book.rentals.last.person).to eq(person)
  end
end
```

File: spec/person_spec.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  end
+
+  describe '#can_use_services?' do
+    context 'when the person is of age' do
+      it 'returns true' do
+        expect(person.can_use_services?).to be(true)
+      end
+    end
+  end
+end
```
Comment (Whoistolu) [inline spec/person_spec.rb]: - [ ] [REQUIRED] The other methods (`create_persons` and `of_age`) have not been tested in these provided unit tests so far. You need to write additional tests for them, similar to how the existing methods are tested. This is a part of the project requirement.

File: spec/student_spec.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+      expect(student.classroom).to eq('Math')
+    end
+  end
+
+  context 'play_hooky' do
+    it 'returns a shrug emoticon' do
+      expect(student.play_hooky).to eq('¯\(ツ)/¯')
+    end
+  end
+end
```
Comment (Whoistolu) [inline spec/student_spec.rb]: - [ ] [REQUIRED] Kindly add a unit test for the `classroom` method in the student class as well. For example:

``` ruby
describe Student do
  let(:student) { Student.new(16, 'Alice', parent_permission: true, classroom: 'Math') }
  let(:classroom) { Classroom.new('Math') }

  context 'classrooms=' do
    it 'assigns the classroom and adds the student to the classroom' do
      student.classrooms = classroom
      expect(student.classroom).to eq(classroom)
      expect(classroom.students).to include(student)
    end
  end
end
```

**Review by Whoistolu — CHANGES_REQUESTED** [review]
Hi Team,
@boniface193 and @prg-04 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

### Highlight
🔆 There are some unit tests for some of the methods in all the class

## Required Changes ♻️

- [ ]  _Check the inline comments under the review for sections to improve._

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

### N/A

Cheers, and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by prolajumokeoni — APPROVED** [review]
Hi @boniface193 and @prg-04 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
 - Tests  requested have been added  ✅ 
   

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**